quinn-proto/src/connection/mod.rs

```diff
@@ -1319,12 +1335,29 @@ impl Connection {
 
             // Send an off-path PATH_RESPONSE. Prioritized over on-path data to ensure that
             // path validation can occur while the link is saturated.
-            if space_id == SpaceId::Data && builder.buf.num_datagrams() == 1 {
+            if space_id == SpaceId::Data && builder.buf.is_empty() {
```

Comment:
I know we spoke about this earlier and assumed this was a better check. But reviewing now I can see that we've already created a `PacketBuilder` on line 1230. At that time we already wrote a packet header into the transmit buffer. So really the previous check was correct because we should be in the first packet being built.